extensions/interactions/PencilCodeEditor/directives/pencil-code-editor-validation.service.spec.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for Pencil Code Editor Validation Service.
+ */
+import { TestBed } from '@angular/core/testing';
+
+import { OutcomeObjectFactory } from
+  'domain/exploration/OutcomeObjectFactory';
+
+import { PencilCodeEditorValidationService } from
+  // eslint-disable-next-line max-len
+  'interactions/PencilCodeEditor/directives/pencil-code-editor-validation.service';
+import { RuleObjectFactory, RuleInputs } from
+  'domain/exploration/RuleObjectFactory';
+import { AnswerGroupObjectFactory } from
+  'domain/exploration/AnswerGroupObjectFactory';
+import { AppConstants } from 'app.constants';
+
+
+describe('Pencil Code Editor Validation Service', () => {
+  let pcevs: PencilCodeEditorValidationService = null;
+  let oof: OutcomeObjectFactory = null;
+  let rof: RuleObjectFactory = null;
+  let inputBackend: RuleInputs = null;
+  let agof : AnswerGroupObjectFactory = null;
+
+  beforeEach(() => {
+    oof = TestBed.get(OutcomeObjectFactory);
+    pcevs = TestBed.get(PencilCodeEditorValidationService);
+    rof = TestBed.get(RuleObjectFactory);
+    agof = TestBed.get(AnswerGroupObjectFactory);
+  });
+
+  it('should return empty list', () => {
```

Comment:
`should return empty list` when?

In general, the describeMessage + itMessage should be a complete sentence. (Over here it's `'Pencil Code Editor Validation Service should return empty list'` follow this pattern in other place.

Reply:
done